patches/api/0114-Add-EntityKnockbackByEntityEvent-and-EntityPushedByE.patch

```diff
@@ -59,6 +69,58 @@ index 0000000000000000000000000000000000000000..e0ba692c9b107f2b042a9c06549185e1
 +        return super.getPushedBy();
 +    }
 +
++    /**
++     * Gets the cause of the knockback.
++     *
++     * @return the cause of the knockback
++     */
++    @NotNull
++    public KnockbackCause getCause() {
++        return cause;
++    }
++
++    /**
++     * An enum to specify the cause of the knockback.
++     */
++    public enum KnockbackCause {
++
++        /**
++         * Knockback caused by non-entity damage.
++         */
++        DAMAGE,
++        /**
++         * Knockback caused by an attacking entity.
++         */
++        ENTITY_ATTACK,
++        /**
++         * Knockback caused by an explosion.
++         */
++        EXPLOSION,
++        /**
++         * Knockback caused by the target blocking with a shield.
++         */
++        SHIELD_BLOCK,
++        /**
++         * Knockback caused by a sweeping attack.
++         */
++        SWEEP_ATTACK,
++        /**
++         * Knockback with an unknown cause.
++         */
++        UNKNOWN;
++    }
++
++    public static KnockbackCause convertNewCauseToOldCause(EntityKnockbackEvent.KnockbackCause knockbackCause) {
```

Comment:
This should be moved into the server module to avoid being exposed into the api. Maybe in CraftEventFactory?

Reply:
I'd much prefer we just follow in terms of adding it to the method which fires that events params, that way it's easier to strip the spigot one out in due course